apis/v1alpha2/httproute_types.go

```diff
@@ -641,6 +657,35 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// Protocol is the protocol to be used in the value of the `Location`
+	// header in the response.
+	// When empty, the protocol of the request is used.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=HTTP;HTTPS
+	Protocol *string `json:"protocol,omitempty"`
+	// Hostname is the hostname to be used in the value of the `Location`
+	// header in the response.
+	// When empty, the hostname of the request is used.
+	//
+	// +optional
+	Hostname *string `json:"hostname,omitempty"`
+	// Port is the port to be used in the value of the `Location`
+	// header in the response.
+	// When empty, port (if specified) of the request is used.
+	//
+	// +optional
+	Port *int `json:"port,omitempty"`
+	// StatusCode is the HTTP status code to be used in response.
+	//
+	// +optional
+	// +kubebuilder:default=302
+	// +kubebuilder:validation=301;302
+	StatusCode *int `json:"redirect_status_code,omitempty"`
```

Comment:
Sorry to bring this up _after_ the merge, but should the json tag be lower camel-case?
```suggestion
	StatusCode *int `json:"statusCode,omitempty"`
```